CRM/Activity/ActionMapping.php

```diff
@@ -108,7 +108,7 @@ public function createQuery($schedule, $phase, $defaultParams) {
     $query['casDateField'] = 'e.activity_date_time';
 
     if (!is_null($schedule->limit_to)) {
-      $activityContacts = \CRM_Core_OptionGroup::values('activity_contacts', FALSE, FALSE, FALSE, NULL, 'name');
+      $activityContacts = \CRM_Activity_BAO_ActivityContact::buildOptions('record_type_id', 'validate');
```

Comment:
Putting this here for easy reference
```
 public static function &values(
    $name, $flip = FALSE, $grouping = FALSE,
    $localize = FALSE, $condition = NULL,
    $labelColumnName = 'label', $onlyActive = TRUE, $fresh = FALSE, $keyColumnName = 'value',
    $orderBy = 'weight')
```
Here is the definition of the contexts for buildOptions
```
    $contexts = array(
      'get' => "get: all options are returned, even if they are disabled; labels are translated.",
      'create' => "create: options are filtered appropriately for the object being created/updated; labels are translated.",
      'search' => "search: searchable options are returned; labels are translated.",
      'validate' => "validate: all options are returned, even if they are disabled; machine names are used in place of labels.",
      'abbreviate' => "abbreviate: enabled options are returned; labels are replaced with abbreviations.",
      'match' => "match: enabled options are returned using machine names as keys; labels are translated.",
    );
```

In the case of this change the $onlyActive is TRUE in the existing code & FALSE in the changed version, but I don't find it valid that we should be disabling ActivityContact record types.

Definition of record_type_id DOES refer to activity_contacts
```
  <field>
    <name>record_type_id</name>
    <type>int unsigned</type>
    <title>Record Type ID</title>
    <comment>Nature of this contact's role in the activity: 1 assignee, 2 creator, 3 focus or target.</comment>
    <pseudoconstant>
      <optionGroupName>activity_contacts</optionGroupName>
    </pseudoconstant>
    <html>
      <type>Select</type>
    </html>
    <add>4.4</add>
  </field>
```

Summary of above - change looks good :-)